include/layers/FCLayer.hpp

```diff
@@ -1,17 +1,64 @@
 #pragma once
+#include <algorithm>
+#include <initializer_list>
+#include <numeric>
 #include <stdexcept>
 #include <vector>
 
+// height -> width -> other dims
```

Comment:
Not necessary this layout. Height and width could be not the first two dims, it's better to remove this misleading comment
```suggestion
```